completion/handlers.go

```diff
@@ -0,0 +1,75 @@
+package completion
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/coder/serpent"
+)
+
+// FileHandler returns a handler that completes files, using the
+// given filter func, which may be nil.
+func FileHandler(filter func(info os.FileInfo) bool) serpent.CompletionHandlerFunc {
+	return func(inv *serpent.Invocation) []string {
+		return ListFiles(inv.CurWord, filter)
+	}
+}
+
+func FileListHandler(filter func(info os.FileInfo) bool) serpent.CompletionHandlerFunc {
+	return func(inv *serpent.Invocation) []string {
+		curWord := strings.TrimLeft(inv.CurWord, `"`)
+		if curWord == "" {
+			return ListFiles("", filter)
+		}
+		parts := strings.Split(curWord, ",")
+		out := ListFiles(parts[len(parts)-1], filter)
+		// prepend := strings.Join(parts[:len(parts)-1], ",")
+		for i, s := range out {
+			parts[len(parts)-1] = s
+			out[i] = strings.Join(parts, ",")
+		}
+		return out
+	}
+}
+
+func ListFiles(word string, filter func(info os.FileInfo) bool) []string {
+	out := make([]string, 0, 32)
```

Comment:
32 initial cap seems arbitrary?

if there's a good reason it should become a comment, otherwise make this `var out []string` to reduce cognitive overhead.

Reply:
Yeah, it's arbitrary, I just wanted to avoid allocing the array 5 times for 16 files (Assuming `var out []string` gives us a zero capacity: 0->1->2->4->8->16)